stdlib/in_channel.mli

```diff
@@ -0,0 +1,161 @@
+(*                                                                        *)
+(*   Copyright 2019 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+
+(** Input channels.
+
+    @since 4.10 *)
+
+type t = in_channel
+
+val stdin : t
+(** Standard output. *)
+
+(** {2 General input functions} *)
+
+type 'a open_function =
+  ?create:bool -> (* default: false *)
+  ?binary:bool -> (* default: false *)
+  'a
+
+val create : t open_function
+(** TBA *)
+
+val with_file : (string -> (t -> 'a) -> 'a) open_function
+(** TBA *)
+
+val close : t -> unit
+(** Close the given channel.  Input functions raise a [Sys_error]
+    exception when they are applied to a closed input channel,
+    except [close_in], which does nothing when applied to an already
+    closed channel. *)
+
+val input : t -> bytes -> int -> int -> int
+(** [input ic buf pos len] reads up to [len] characters from
+    the given channel [ic], storing them in byte sequence [buf], starting at
+    character number [pos].
+    It returns the actual number of characters read, between 0 and
+    [len] (inclusive).
+    A return value of 0 means that the end of file was reached.
+    A return value between 0 and [len] exclusive means that
+    not all requested [len] characters were read, either because
+    no more characters were available at that time, or because
+    the implementation found it convenient to do a partial read;
+    [input] must be called again to read the remaining characters,
+    if desired.  (See also {!Stdlib.really_input} for reading
+    exactly [len] characters.)
+    Exception [Invalid_argument "input"] is raised if [pos] and [len]
+    do not designate a valid range of [buf]. *)
+
+val really_input : t -> bytes -> int -> int -> unit option
+(** [really_input ic buf pos len] reads [len] characters from channel [ic],
+    storing them in byte sequence [buf], starting at character number [pos].
+    Return [None] if the end of file is reached before [len]
+    characters have been read.
+    Raise [Invalid_argument "really_input"] if
+    [pos] and [len] do not designate a valid range of [buf]. *)
+
+val really_input_string : t -> int -> string option
+(** [really_input_string ic len] reads [len] characters from channel [ic]
+    and returns them in a new string.
+    Return [None] if the end of file is reached before [len]
+    characters have been read. *)
+
```

Comment:
`really_X` is a terrible naming scheme IMO. Can we come up with a better naming scheme? How about `input_n` and `input_string_n`?

Reply:
I'm not married to the existing name, but `input_n` is not any better IMO.

Reply:
I'm not married to the existing name, but `input_n` is not any better IMO.

Reply:
Let me make the case for it: it mirrors C's `printf`/`snprintf` scheme ie. it succinctly signifies that there is something different about the function, and it can easily be remembered that the difference concerns the number of input elements.

Reply:
please don't emulate anything from C's IO, it's terrible :p
I think Lwt has `input_exactly` or something to this taste, which is maybe more. I think `input_n` doesn't mean anything, and is worse than `really_input` which at least is familiar.

Reply:
`input_exactly` sounds a lot better than `really_input`. `really_input` sounds really... bad.